__tests__/registries/npm-registry.js

```diff
@@ -731,6 +731,37 @@ describe('isScopedPackage functional test', () => {
   });
 });
 
+describe('environment variables functional test', () => {
+  beforeEach(() => {
+    process.env.npm_config_always_auth = 'true';
+    process.env.npm_config__auth = 'auth';
+    process.env.npm_config__authtoken = 'authToken';
+    process.env.npm_config__username = 'username';
+    process.env.npm_config__password = 'password';
+  });
+
+  afterEach(() => {
+    delete process.env.npm_config_always_auth;
+    delete process.env.npm_config__auth;
+    delete process.env.npm_config__authToken;
+    delete process.env.npm_config__username;
+    delete process.env.npm_config__password;
+  });
+
+  test('correctly escapes environment config variables', () => {
+    const testCwd = '.';
+    const {mockRequestManager, mockRegistries, mockReporter} = createMocks();
+    const npmRegistry = new NpmRegistry(testCwd, mockRegistries, mockRequestManager, mockReporter, true, []);
```

Comment:
I had trouble figuring out where these tests should go. `mergeEnv` is an instance method that requires a whole mocked registry to test.

Simplifying this test would involve refactoring the `mergeEnv` method to use a utility method, but I decided to reduce the changes to the code and instead have a more complicated test